src/constraint.rs

```diff
@@ -534,15 +541,28 @@ impl CoreAction {
                     .chain(get_atom_application_constraints(f, &args, span, typeinfo)?)
                     .collect())
             }
-            CoreAction::Set(span, head, args, rhs) => {
+            CoreAction::Set(span, head, args, rhs, is_cost) => {
```

Comment:
I think it would be cleaner to have a new core action that sets costs rather than re-use `Set` here.

Reply:
Sure

Reply:
I changed to create our own action